### ISSUE TITLE
Psalm doesn't recognise string-only keys after array_key_intersect

### ISSUE THREAD

**Issue by 7ochem** [vimeo/psalm#9202]
If I have an array in my code that can contain anything and intersect it by key with an array that only has string keys, then Psalm doesn't detect that. If I want to pass that array to a function that expects `array<string, mixed>` Psalm is reporting: `INFO: [MixedArgumentTypeCoercion](https://psalm.dev/194) - 19:10 - Argument 1 of consumer expects array<string, mixed>, but parent type array<array-key, mixed> provided`

https://psalm.dev/r/a6a6c1da31

**Comment by 7ochem** [comment]
Ah, never mind. I experimented a bit. PHP manual on array_intersect_key also says:

> The two keys from the key => value pairs are considered equal only if (string) $key1 === (string) $key2 . In other words a strict type check is executed so the string representation must be the same.

```php
function params(): array { return ['id' => 123, 123 => '123']; }
var_dump(array_intersect_key(params(), ['123' => 0, 'id' => 0]));
// Gives:
// array(2) {
//   'id' =>
//   int(123)
//   [123] =>
//   string(3) "123"
// }
```
